nncf/experimental/torch/sparsify_activations/torch_backend.py

```diff
@@ -0,0 +1,198 @@
+#      http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Dict, List, Type, TypeVar
+
+import numpy as np
+import torch
+import torch.nn as nn
+
+import nncf
+from nncf.common.graph.graph import NNCFGraph
+from nncf.common.graph.graph import NNCFNode
+from nncf.common.graph.operator_metatypes import CONST_NOOP_METATYPES
+from nncf.common.graph.operator_metatypes import OperatorMetatype
+from nncf.common.graph.transformations.commands import TargetType
+from nncf.data import Dataset
+from nncf.experimental.torch.sparsify_activations.sparsify_activations_impl import SparsifyActivationsAlgoBackend
+from nncf.torch.graph import operator_metatypes as om
+from nncf.torch.graph.transformations.commands import PTSharedFnInsertionCommand
+from nncf.torch.graph.transformations.commands import PTTargetPoint
+from nncf.torch.graph.transformations.layout import PTTransformationLayout
+from nncf.torch.model_transformer import PTModelTransformer
+from nncf.torch.nncf_network import NNCFNetwork
+from nncf.torch.utils import training_mode_switcher
+
+ACTIVATIONS_SPARSIFIER_PREFIX = "activations_sparsifier"
+TModel = TypeVar("TModel")
+
+
+class ActivationsSparsifier(nn.Module):
+    """
+    Sparsifies input activations by masking out values around zero.
+    """
+
+    def __init__(self, target_sparsity: float, alpha: float = 0.2):
+        """
+        :param target_sparsity: The target activation sparsity level.
+        :param alpha: The exponential moving average decay factor in range (0, 1) for calibrating
+            the threshold. A larger alpha will give more weight to the most recent batches.
+        """
+        super().__init__()
+        self.target_sparsity = target_sparsity
+        if alpha <= 0.0 or alpha >= 1.0:
+            raise ValueError("The decay factor `alpha` should be in range (0, 1).")
+        self.alpha = alpha
+        self.register_buffer("running_threshold", torch.tensor(float("-inf")))
+        self.register_buffer("num_batches_tracked", torch.tensor(0))
+        self.running_threshold: torch.Tensor
+        self.num_batches_tracked: torch.Tensor
+        self._freeze = True
+
+    @staticmethod
+    def calculate_threshold(x: torch.Tensor, target_sparsity: float) -> torch.Tensor:
+        """
+        Calculates the threshold for sparsifying the input tensor if locations of `x.abs() <= threshold` are zeroed.
+
+        :param x: The input tensor.
+        :param target_sparsity: The target sparsity level on the input tensor.
+        :return: The threshold value.
+        """
+        # uses numpy's quantile implementation as torch's cannot handle large tensor
```

Comment:
I believe for tensors smaller than `16_000_000` torch implementation does still work. It also should be faster than numpy if tensor is on CUDA device (not sure about CPU device). Maybe it's worth processing it similar to how it's done [here](https://github.com/openvinotoolkit/nncf/blob/develop/nncf/tensor/functions/torch_numeric.py#L241)?

Reply:
I would suggest to use `quantile` from the `torch.numeric` file here directly

Reply:
changed to use `nncf.tensor.functions.torch_numeric.quantile`